src/pages/DisbursementsNew.tsx

```diff
@@ -155,9 +157,43 @@ export const DisbursementsNew = () => {
     if (apiError) {
       dispatch(clearDisbursementDraftsErrorAction());
     }
+    calculateDisbursementTotalAmountFromFile(file);
     setCsvFile(file);
   };
 
+  const calculateDisbursementTotalAmountFromFile = (file?: File) => {
+    if (file) {
+      const reader = new FileReader();
+      reader.readAsText(file);
+      const handleLoadFile = () => {
+        const totalAmount = reader.result
+          ?.toString()
+          .split("\n")
+          .slice(1)
+          .reduce(
+            (accumulator, line) =>
+              !line ? accumulator : accumulator + Number(line.split(",")[2]),
+            0,
+          );
+
+        setDraftDetails({
+          ...draftDetails,
+          stats: {
+            ...draftDetails?.stats,
+            totalAmount: totalAmount?.toString() ?? "0",
+          },
+        } as Disbursement);
+
+        // update future balance
+        const assetBalance = allBalances?.find(
+          (a) => a.assetCode === draftDetails?.asset.code,
+        )?.balance;
+        setFutureBalance(Number(assetBalance) - totalAmount!);
```

Comment:
It might be safer to check that we have the `totalAmount` instead of assuming it will be there. What do you think?

```suggestion
     if (totalAmount) {
        setFutureBalance(Number(assetBalance) - totalAmount);
      }
```